src/events/api/attendee.ts

```diff
@@ -1,12 +1,63 @@
-import { get } from 'common/utils/api';
-import { IAttendee } from 'events/models/Attendee';
+import { get, post, deleteR } from 'common/utils/api';
 
 import { getUser } from 'authentication/api';
 
+import { IAttendee } from 'events/models/Attendee';
+import { IAuthUser } from 'authentication/models/User';
+
 const getEventAttendeeUrl = (eventId: number) => `/api/v1/event/attendance-events/${eventId}/attendee/`;
+const getUserAttendUrl = (eventId: number) => `/api/v1/event/attendance-events/${eventId}/register/`;
+const getUserUnAttendUrl = (eventId: number) => `/api/v1/event/attendance-events/${eventId}/unregister/`;
 
-export const getAttendeeForEvent = async (eventId: number) => {
+interface IAttendEventOptions {
+  allowPictures?: boolean;
+  showAsAttending?: boolean;
+  extras?: number[]; // options id
+}
+
+export const getAttendeeForEvent = async (eventId: number): Promise<IAttendee> => {
   const user = await getUser();
   const attendee = await get<IAttendee>(getEventAttendeeUrl(eventId), { format: 'json' }, { user });
   return attendee;
 };
+
+export const userAttendEvent = async (
+  event_id: number,
```

Comment:
I think we should use camel casing until we get at far down to the API as possible

--
Maybe we should automatically transform all snake casing for requests at a later time?